web/vim_lib.js

```diff
@@ -72,6 +75,38 @@ var LibraryVIM = {
 
     dropbox: null,
 
+    // from https://github.com/s-macke/jor1k/blob/master/js/lib/utf8.js
+    // LICENSE: https://github.com/s-macke/jor1k/blob/master/LICENSE.md
+    UnicodeToUTF8Stream: function(key) {
```

Comment:
Ok, I made it work using emscripten UTF8 functions. The problem was just with null termination.